src/component/elements/popup/Modal/Provider.tsx

```diff
@@ -77,17 +77,18 @@ function Provider({
     },
     [setModal],
   );
-
+  // todo: optimize this
+  const wrapperRect = wrapperRef?.getBoundingClientRect();
   const parentStyle = useMemo(() => {
```

Comment:
I'm pretty sure you can remove the `useMemo`, because `wrapperRect` is always a different object.

Reply:
I added wrapperRect, because using wrapperRef as dependency doesn't work correctly (the prev version)
and i found this solution
https://gist.github.com/morajabi/523d7a642d8c0a2f71fcfa0d8b3d2846